mautic/mautic
# Replacing unmaintained hard-coded Less library with a better option

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

Replacing https://github.com/wikimedia/less.php with https://github.com/wikimedia/less.php

Installing the new library using Composer instead of hard-coding it to the Mautic's codebase. The old one had warnings in PHP 8.1 and so this is needed for PHP 8.1 support (https://github.com/mautic/mautic/pull/12382)

The deprecation this is solving: 
```
PHP Deprecated:  preg_match(): Passing null to parameter #4 ($flags) of type int is deprecated in /home/runner/work/mautic/mautic/plugins/MauticFocusBundle/Include/lessc.inc.php on line 3996
```

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Test that the Focus items can modify the view of the modal or bar.
